lib/vyasa_web/live/display_manager/display_live.ex

```diff
@@ -0,0 +1,531 @@
+     |> stream_insert(
+       :verses,
+       %{verses[curr_verse_id] | binding: nil}
+     )
+     |> stream_insert(
+       :verses,
+       %{verses[verse_id] | binding: bind}
+     )
+     |> assign(:marks, [%{d_mark | binding: bind, verse_id: verse_id} | marks])}
+  end
+
+  # already in mark in drafting state, remember to late bind binding => with a fn()
+  def handle_event(
+        "bindHoveRune",
+        %{"binding" => bind = %{"verse_id" => verse_id}},
+        %{assigns: %{kv_verses: verses, marks: [%Mark{state: :draft} = d_mark | marks]}} = socket
+      ) do
+    # binding here blocks the stream from appending to quote
+    bind = Draft.bind_node(bind)
+
+    {:noreply,
+     socket
+     |> stream_insert(
+       :verses,
+       %{verses[verse_id] | binding: bind}
+     )
+     |> assign(:marks, [%{d_mark | binding: bind, verse_id: verse_id} | marks])}
+  end
+
+  @impl true
+  def handle_event(
+        "bindHoveRune",
+        %{"binding" => bind = %{"verse_id" => verse_id}},
+        %{assigns: %{kv_verses: verses, marks: [%Mark{order: no} | _] = marks}} = socket
+      ) do
+    bind = Draft.bind_node(bind)
+
+    {:noreply,
+     socket
+     |> stream_insert(
+       :verses,
+       %{verses[verse_id] | binding: bind}
+     )
+     |> assign(:marks, [
+       %Mark{state: :draft, order: no + 1, verse_id: verse_id, binding: bind} | marks
+     ])}
+  end
+
+  @impl true
+  def handle_event(
+        "markQuote",
+        _,
+        %{assigns: %{marks: [%Mark{state: :draft} = d_mark | marks]}} = socket
+      ) do
+    IO.inspect(marks)
+    {:noreply, socket |> assign(:marks, [%{d_mark | state: :live} | marks])}
+  end
+
+  def handle_event("markQuote", _, socket) do
+    {:noreply, socket}
+  end
+
+  def handle_event(
+        "createMark",
+        %{"body" => body},
+        %{assigns: %{marks: [%Mark{state: :draft} = d_mark | marks]}} = socket
+      ) do
+    {:noreply, socket |> assign(:marks, [%{d_mark | body: body, state: :live} | marks])}
+  end
+
+  def handle_event("createMark", _event, socket) do
+    {:noreply, socket}
+  end
+
+  def handle_event(event, message, socket) do
+    IO.inspect(%{event: event, message: message}, label: "pokemon")
+    {:noreply, socket}
+  end
+
+  # CHECKPOINT: handle_info functions in DM
+  @doc """
+  Handles the custom message that corresponds to the :media_handshake event with the :init
+  message, regardless of the module that dispatched the message.
+
+  This indicates an intention to sync the media library with the chapter, hence it
+  returns a message containing %Voice{} info that can be used to generate a playback struct.
+  """
+  @impl true
+  def handle_info(
+        {_, :media_handshake, :init} = _msg,
+        %{
+          assigns: %{
+            session: %{"id" => sess_id},
+            chap: %Chapter{no: c_no, source_id: src_id}
+          }
+        } = socket
+      ) do
+    %Voice{} = chosen_voice = Medium.get_voice(src_id, c_no, @default_voice_lang)
```

Comment:
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]** (MatchError) no match of right hand side value: nil
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]    (vyasa 0.1.0-alpha.1) lib/vyasa_web/live/display_manager/display_live.ex:489: VyasaWeb.DisplayManager.DisplayLive.handle_info/2
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]    (phoenix_live_view 0.20.17) lib/phoenix_live_view/channel.ex:360: Phoenix.LiveView.Channel.handle_info/2
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]    (stdlib 5.1.1) gen_server.erl:1077: :gen_server.try_handle_info/3
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]    (stdlib 5.1.1) gen_server.erl:1165: :gen_server.handle_msg/6
2024-08-16T17:16:08Z app[3d8d9564b91258] sin [info]    (stdlib 5.1.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3

Reply:
found rootcause it happens when we move to chapters without voices as of yet and throws nil

Reply:
```suggestion
        case Medium.get_voice(src_id, c_no, @default_voice_lang) do
      %Voice{} = v ->
        Vyasa.PubSub.publish(
          v,
          :voice_ack,
          sess_id
        )

        _ - > nil
    end

```

Reply:
wonderful, thanks for this